frontend/src/components/custom/UserOptionsModal.tsx

```diff
@@ -0,0 +1,221 @@
+import { ChevronDownIcon, ChevronUpIcon } from '@chakra-ui/icons';
+import {
+  Button,
+  Menu,
+  MenuButton,
+  MenuItem,
+  MenuList,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from '@chakra-ui/react';
+import React, { useState } from 'react';
+
+interface UserOptionsModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+interface Report {
```

Comment:
Rather than being declared here, this data interface should probably be moved to the [models](https://github.com/TalkToBeavs/TalkToBeavs/tree/deploy/models) directory since the submitted reports will be stored in MongoDB. However, it's fine to leave it as is for now. We just need to remember to refactor it once we implement the actual backend logic for storing the submitted reports and sending notifications about them to the admin.

Reply:
Agreed that it should be moved into the models directory, I believe my thought process for the time being was just having it there for the time being until we create the models for prisma and then learning how to integrate with that since we are switching to postgresql. I will be sure to move the all data interfaces to this directory upon implementation. Thanks for the feedback 😄